src/app/app.component.ts

```diff
@@ -12,14 +12,18 @@ export class AppComponent {
     alert("I did stuff");
   }
 
-  doOtherStuff() {
-    try {
-      for (let i = 0; i < 50; i++) {
-        console.log('looping for no good reason');
-      }
-      alert("I did other stuff");
-    } catch (e) {
-      throw e;
+  public doOtherStuff() {
+    var a = [1, 2, 3];
+    var b = a.reverse(); // Noncompliant
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Move this array "reverse" operation to a separate statement. ([typescript:S4043](http://localhost:9076/coding_rules#rule_key=typescript%3AS4043))

[See it in SonarQube](http://localhost:9076/project/issues?id=org.sonarqube%3Aangular-sample&pullRequest=1&issues=AWmQHKGAZQKnx6JGnxzp&open=AWmQHKGAZQKnx6JGnxzp)

[ISSUE_KEY]: AWmQHKGAZQKnx6JGnxzp